hunts/src/puzzle-table.js

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { useTable, useExpanded } from "react-table";
+import { useGlobalFilter } from "./use-global-filter-fixed.js";
+import { matchSorter } from "match-sorter";
+
+export function PuzzleTable({ columns, data }) {
+    return usePuzzleTable({
+        columns: React.useMemo(() => columns, [columns]),
+        data: React.useMemo(() => data, [data]),
+    });
+}
+
+PuzzleTable.propTypes = {
+    columns: PropTypes.array.isRequired,
+    data: PropTypes.array.isRequired,
+};
+
+function textFilterFn(rows, id, filterValue) {
+    if (!filterValue || !filterValue.length) {
+        return rows;
+    }
+
+    const words = filterValue.split(' ');
+    if (!words) {
+        return rows;
+    }
+
+    const keys = ["values.name", "values.tags", "values.status", "values.answer"];
+    // Need to clone these results because this library is broken as shit
+    return words.reduceRight(
+        (results, word) => matchSorter(results, word, { keys }),
+        rows).map(row => Object.assign({}, row));
+}
+
+textFilterFn.autoRemove = val => !val;
+
+function usePuzzleTable({ columns, data }) {
+    const filterTypes = React.useMemo(
+        () => ({
+            globalFilter: textFilterFn,
+        }),
+        []
+    );
+
+    const {
+        getTableProps,
+        getTableBodyProps,
+        allColumns,
+        rows,
+        state,
+        prepareRow,
+        toggleAllRowsExpanded,
+        flatRows,
+        preGlobalFilteredRows,
+        setGlobalFilter,
+    } = useTable(
+        {
+            columns,
+            data,
+            filterTypes,
```

Comment:
Having read through this PR super thoroughly, I'm pretty confident we could cut a lot of complexity by home-rolling our filtering and _probably_ our nesting capabilities.

Fundamentally, our table state is calculable from our data, our filter, and our collapsed nodes, it looks like a lot of this code is fighting `react-table`'s default behaviors, I think we can largely avoid it.

Reply:
I'm actually not sure all that much of it right now is fighting react-table, I think only a couple lines in this file are doing that right now (but the `-fixed.js` file changes definitely do fall in that bucket). At this point I have little affection for this library but I'm not sure if it's worth writing our own version.